app/controllers/reports_controller.rb

```diff
@@ -2,13 +2,13 @@ class ReportsController < ApplicationController
   def create
     report = Report.new (report_params)
     if report.save 
-      flash[:notice] = "Post reported"
-      respond_to do |format|
-        format.js
-        format.html {  redirect_to post_path(report.post_id, report.post) }
-      end
+      flash[:notice] = "Post is successfully reported"
     else
-      render "form"
+      flash[:alert] = "Post is not reported"
```

Comment:
This error message can also be improved.

"Sorry something went wrong and this post was not reported. Can you please try again?"

Reply:
Here also the message is improved